src/index.js

```diff
@@ -1,24 +1,30 @@
-'use strict';
+import { JSDOM } from 'jsdom';
 
-import dom from 'jsdom';
+const SGPC_URL = 'http://old.sgpc.net/hukumnama/sgpconlinehukamnama.asp';
 
-const SGPC_URL = `http://old.sgpc.net/hukumnama/sgpconlinehukamnama.asp`;
+module.exports = () => new Promise((resolve, reject) => {
+  JSDOM.fromURL(SGPC_URL)
+    .then(dom => {
+      const { document } = dom.window;
 
-module.exports = (options = {  }) => new Promise((resolve, reject) => dom.env(SGPC_URL, (err, { document }) => {
-  try { 
-    const cleanTable = table => table.textContent.replace(/[\t\r\n]/g, '').trim();
-    const tables = [...document.querySelectorAll('table')].splice(1).map(cleanTable);
-    const [ gurbani, gurbaniAng, punjabi, english, englishAng ] = tables;
+      try {
+        const cleanTable = table => table.textContent.replace(/[\t\r\n]/g, '').trim();
+        const tables = [...document.querySelectorAll('table')].splice(1).map(cleanTable);
+        // eslint-disable-next-line no-unused-vars
+        const [gurbani, gurbaniAng, punjabi, english, englishAng] = tables;
 
-    resolve({
-      content: tables.join('\n'),
-      gurbani, punjabi, english,
-      ang: parseInt(/Page:[ ]*([\d ]+)/.exec(englishAng)[1]),
-      date: document.querySelector('font[face="Georgia, Times New Roman, Times, serif"]').textContent.trim().slice(1,-1),
-      url: SGPC_URL
-    });
-
-  } catch (e) {
-    reject(e);
-  }
-}));
+        resolve({
+          content: tables.join('\n'),
+          gurbani,
+          punjabi,
+          english,
+          ang: parseInt(/Page:[ ]*([\d ]+)/.exec(englishAng)[1], 10),
+          date: document.querySelector('font[face="Georgia, Times New Roman, Times, serif"]').textContent.trim().slice(1, -1),
+          url: SGPC_URL
+        });
+      } catch (e) {
+        reject(e);
+      }
+    })
+    .catch(e => reject);
+});
```

Comment:
If you feel like, you can convert this into a `async` function as node 8 supports it

Reply:
Achhaa Jee